src/internal/packager/images/pull.go

```diff
@@ -168,3 +212,63 @@ func (i *ImgConfig) PullImage(src string, spinner *message.Spinner) (img v1.Imag
 
 	return img, nil
 }
+
+// IndexJSON represents the index.json file in an OCI layout.
+type IndexJSON struct {
+	SchemaVersion int `json:"schemaVersion"`
+	Manifests     []struct {
+		MediaType   string            `json:"mediaType"`
+		Size        int               `json:"size"`
+		Digest      string            `json:"digest"`
+		Annotations map[string]string `json:"annotations"`
+	} `json:"manifests"`
+}
+
+// addImageNameAnnotation adds an annotation to the index.json file so that the deploying code can figure out what the image tag <-> digest shasum will be.
+func addImageNameAnnotation(ociPath string, digestToTag map[string]string) error {
+	indexPath := filepath.Join(ociPath, "index.json")
+
+	// Add an 'org.opencontainers.image.base.name' annotation so we can figure out what the image tag/digest shasum will be during deploy time
+	indexJSON, err := os.Open(indexPath)
+	if err != nil {
+		message.Errorf(err, "Unable to open %s/index.json", ociPath)
+		return err
+	}
+
+	// Read the file contents and turn it into a useable struct that we can manipulate
+	var index IndexJSON
+	byteValue, err := io.ReadAll(indexJSON)
+	if err != nil {
+		return fmt.Errorf("unable to read the contents of the file (%s) so we can add an annotation: %w", indexPath, err)
+	}
+	indexJSON.Close()
+	if err = json.Unmarshal(byteValue, &index); err != nil {
+		return fmt.Errorf("unable to process the conents of the file (%s): %w", indexPath, err)
+	}
+	for idx, manifest := range index.Manifests {
+		if manifest.Annotations == nil {
+			manifest.Annotations = make(map[string]string)
+		}
+		manifest.Annotations[ocispec.AnnotationBaseImageName] = digestToTag[manifest.Digest]
+		index.Manifests[idx] = manifest
+	}
+
+	// Remove any file that might already exist
+	_ = os.Remove(indexPath)
+
+	// Create the index.json file and save the data to it
+	indexJSON, err = os.Create(indexPath)
+	if err != nil {
+		return err
+	}
+	indexJSONBytes, err := json.Marshal(index)
+	if err != nil {
+		return err
+	}
+	_, err = indexJSON.Write(indexJSONBytes)
+	if err != nil {
+		return err
+	}
```

Comment:
This seems worthy of a utils function (similar to our yaml file but for json files).